src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -31,13 +32,21 @@ private static string GetSolutionFolderPathWithForwardSlashes(string path)
             return "/" + string.Join("/", PathUtility.GetPathWithDirectorySeparator(path).Split(Path.DirectorySeparatorChar, StringSplitOptions.RemoveEmptyEntries)) + "/";
         }
 
+        private static bool IsSolutionFolderPathInDirectoryScope(string relativePath)
+        {
+            return !string.IsNullOrEmpty(relativePath)
```

Comment:
So for instance, if I had a path like "src\Build\..\..\..\..\test\foo"...

Reply:
I am not sure it is entirely possible to run into this issue since `relativePath` is obtained through `Path.GetRelativePath`, which should return the actual resolved path.
So instead of `src\Build\..\..\..\..\test\foo`, it should be `..\..\test\foo`

Reply:
Fair enough; missed that